dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -82,20 +86,31 @@
   public static final String TEST_NAME_GLOBAL_UPPER_BOUND = "Global Upper Bounds";
   public static final String TEST_NAME_DEPENDENCY_CONVERGENCE = "Dependency Convergence";
 
-  public static void main(String[] args)
-      throws IOException, TemplateException, RepositoryException, URISyntaxException {
-
-    Path output = generate();
+  /**
+   * Generates a code hygiene dashboard for a BOM. The file name (pom.xml) of the BOM is specified
+   * as the argument.
+   */
+  public static void main(String[] arguments)
+      throws IOException, TemplateException, RepositoryException, URISyntaxException,
+          PlexusContainerException, ComponentLookupException, ProjectBuildingException {
+    if (arguments.length != 1) {
+      System.err.println("Please specify pom.xml for a BOM.");
+      return;
+    }
+    String bomFile = arguments[0];
+    Path output = generate(Paths.get(bomFile));
     System.out.println("Wrote dashboard into " + output.toAbsolutePath());
   }
 
-  public static Path generate()
-      throws IOException, TemplateException, RepositoryException, URISyntaxException {
-
-    // TODO should pass in maven coordinates as argument
-    DefaultArtifact bom =
-        new DefaultArtifact("com.google.cloud:cloud-oss-bom:pom:1.0.0-SNAPSHOT");
-    List<Artifact> managedDependencies = RepositoryUtility.readBom(bom);
+  @VisibleForTesting
+  static Path generate(Path bomFile)
```

Comment:
We might want a version that takes coordinates too.

Reply:
Added `generate(String)`